Description: Fix some spurious test failures
Author: Jelmer Vernooij <jelmer@debian.org>
Status: Not forwarded upstream (workaround)

Index: unstable/breezy/tests/test_test_server.py
===================================================================
--- unstable.orig/breezy/tests/test_test_server.py
+++ unstable/breezy/tests/test_test_server.py
@@ -77,7 +77,12 @@ class TCPClient(object):
         return self.sock.sendall(s)
 
     def read(self, bufsize=4096):
-        return self.sock.recv(bufsize)
+        try:
+            return self.sock.recv(bufsize)
+        except socket.error as e:
+            if e.errno == errno.ECONNRESET:
+                return b""
+            raise
 
 
 class TCPConnectionHandler(socketserver.BaseRequestHandler):
